pkg/algorithm/cell_allocation.go

```diff
@@ -78,6 +79,70 @@ func buddyAlloc(
 	return false
 }
 
+// after buddyAlloc failed to allocate cells,
+// try to split a higher level cell to get current level cells.
+func splitAlloc(
+	cell *cellBindingPathVertex,
+	freeList ChainCellList,
+	freeCellNum map[CellLevel]int32,
+	currentLevel CellLevel,
+	suggestedNodes common.Set,
+	ignoreSuggestedNodes bool,
+	bindings map[api.CellAddress]*PhysicalCell) bool {
+
+	var splittableCell Cell
```

Comment:
Seems we don't need "var splittbleCell cell" explicitly. Instead, we can just use an integer indicating number of children of a higher-level cell.

Reply:
seems there's no way to get the number without recursing a cell

Reply:
I mean, we can just set HUC[i] = len(freeList[i][0].GetChildren()) when iterating each level. No need for splittableCell = freeList[i][0] then get its children number.
Anyway, just slightly different :)

Reply:
what if there're 10 levels of cells and only level 5 has one free cell, freeList is empty for any i != 5.
there's no good way to calculate such HUC unless passing another fullList.